validation/hooks.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	tap "github.com/mndrix/tap-go"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	var output string
+	config := util.LifecycleConfig{
+		Actions: util.LifecycleActionCreate | util.LifecycleActionStart | util.LifecycleActionDelete,
+		PreCreate: func(r *util.Runtime) error {
+			r.SetID(uuid.NewV4().String())
+			g := util.GetDefaultGenerator()
+			output = filepath.Join(r.BundleDir, g.Spec().Root.Path, "output")
+			prestart1 := rspec.Hook{
+				Path: filepath.Join(r.BundleDir, g.Spec().Root.Path, "/bin/sh"),
```

Comment:
You use this `Join` a lot in this test.  Maybe store it in a temporary variable:

```go
shPath := filepath.Join(r.BundleDir, g.Spec().Root.Path, "/bin/sh")
…
  Path: shPath,
```

Reply:
updated, thanks.